beacon_node/lighthouse_network/src/behaviour/mod.rs

```diff
@@ -868,11 +892,39 @@ impl<TSpec: EthSpec> NetworkBehaviourEventProcess<GossipsubEvent> for Behaviour<
                 }
             }
             GossipsubEvent::Subscribed { peer_id, topic } => {
-                if let Some(subnet_id) = subnet_from_topic_hash(&topic) {
-                    self.network_globals
-                        .peers
-                        .write()
-                        .add_subscription(&peer_id, subnet_id);
+                if let Ok(topic) = GossipTopic::decode(topic.as_str()) {
+                    if let Some(subnet_id) = topic.subnet_id() {
+                        self.network_globals
+                            .peers
+                            .write()
+                            .add_subscription(&peer_id, subnet_id);
+                    }
+                    // Try to send the cached messages for this topic
+                    if let Some(msgs) = self.gossip_cache.retrieve(&topic) {
+                        for data in msgs {
+                            let topic_str: &str = topic.kind().as_ref();
+                            match self.gossipsub.publish(topic.clone().into(), data) {
+                                Ok(_) => {
+                                    warn!(self.log, "Gossip message published on retry"; "topic" => topic_str);
+                                    if let Some(v) = metrics::get_int_counter(
+                                        &metrics::GOSSIP_LATE_PUBLISH_PER_MAIN_TOPIC,
```

Comment:
the naming for the metrics implies its on a MAIN_TOPIC, but we are registering these for any topic right?
Should we remove the name MAIN?

Reply:
There are three versions of this, i've just highlighted one as an example here

Reply:
https://github.com/sigp/lighthouse/pull/2964/commits/401434533cda61d3dc8931a61c214953bfc71e39